src/core/ucc_lib.c

```diff
@@ -101,3 +326,16 @@ ucc_status_t ucc_lib_config_modify(ucc_lib_config_h config, const char *name,
     return ucc_config_parser_set_value(config, ucc_lib_config_table, name,
                                        value);
 }
+
+ucc_status_t ucc_finalize(ucc_lib_info_t *lib)
+{
+    int i;
+    ucc_assert(lib->n_libs_opened > 0);
+    ucc_assert(lib->libs);
+    for (i = 0; i < lib->n_libs_opened; i++) {
+        lib->libs[i]->iface->finalize(lib->libs[i]);
```

Comment:
if failure, should i stop trying to finalize other components and return error? or should i still try to call other finalizatoins and return error in the nend?

Reply:
i think that proceed with next finalizations make sense since libs are independent from each other, but what should be the return code if multiple libs finalize fails I'm not sure

Reply:
ERR_NO_MESSAGE? and print error

Reply:
or the first error code observed?

Reply:
returning the first observed error code.